consensus/src/chained_bft/network.rs

```diff
@@ -485,21 +484,11 @@ impl<T: Payload> NetworkTask<T> {
     ) -> failure::Result<()> {
         let req = BlockRetrievalRequest::try_from(request_msg)?;
         debug!("Received block retrieval request {}", req);
-        let (tx, rx) = oneshot::channel();
         let req_with_callback = IncomingBlockRetrievalRequest {
             req,
-            response_sender: tx,
+            response_sender: callback,
         };
-        self.block_request_tx.push(peer_id, req_with_callback)?;
-        let response = rx.await?;
```

Comment:
I think we can test this behavior by writing a test that calls process_request_block() and then validates that if a new message is sent to NetworkTask, it doesn't get stuck.

Reply:
Also, I can't leave a comment in the right place, but on the NetworkTask struct or NetworkTask::start(), it would be helpful to describe the expected semantics that any message processing is expected to complete without any dependencies.  As you explained, the issue was that state sync starts, then a block retrieval request comes and waits for completion (which depends on event_processor to do some work) and the state sync can't finish because the block retrieval is holding up NetworkTask.

Reply:
modified `test_rpc` to test it's non-blocking and verified with master such test'll stuck